tests/lib/services/aiSyntesis/buffer-reduce.subject.unit.ts

```diff
@@ -0,0 +1,79 @@
+import { expect } from 'chai';
+import { catchError, throwError } from 'rxjs';
+import { TestScheduler } from 'rxjs/testing';
+import sinon from 'sinon';
+
+import { BufferedReducerStopException } from '@/lib/services/aiSynthesis/buffer-reduce.exception';
+import { BufferedReducerSubject } from '@/lib/services/aiSynthesis/buffer-reduce.subject';
+
+describe('BufferedReducerSubject', () => {
+  let testScheduler: TestScheduler;
+
+  beforeEach(() => {
+    testScheduler = new TestScheduler((actual, expected) => {
+      expect(actual).to.deep.eq(expected);
+    });
+  });
+
+  it('buffers values, stops buffering if predicate returns true', () => {
+    const predicate = sinon.fake((buffer: number) => buffer < 2);
+    const accumulator = sinon.fake((buffer: number, value: number) => buffer + value);
+
+    const subject = new BufferedReducerSubject<number>(predicate, accumulator);
+
+    testScheduler.run(({ cold, expectObservable }) => {
+      const test$ = cold('a-b-c-|', { a: 1, b: 1, c: 3 });
+
+      expectObservable(subject).toBe('--b-c-|', {
+        b: 2,
+        c: 3,
+      });
+
+      test$.subscribe(subject);
+    });
+  });
+
+  it('does not buffer values if predicate returns false', () => {
+    const predicate = sinon.fake(() => false);
+    const accumulator = sinon.fake((buffer: number, value: number) => buffer + value);
+
+    const subject = new BufferedReducerSubject<number>(predicate, accumulator);
+
+    testScheduler.run(({ cold, expectObservable }) => {
+      const test$ = cold('a-b-c-|', { a: 1, b: 2, c: 3 });
+
+      expectObservable(subject).toBe('a-b-c-|', {
+        a: 1,
+        b: 2,
+        c: 3,
+      });
+
+      test$.subscribe(subject);
+    });
+  });
+
+  it('buffers values, erroring on not found', () => {
+    const predicate = sinon.fake((buffer: number) => {
+      if (buffer >= 3) throw new BufferedReducerStopException();
+      else return true;
+    });
+    const accumulator = sinon.fake((buffer: number, value: number) => buffer + value);
+
+    const subject = new BufferedReducerSubject<number>(predicate, accumulator);
+
+    testScheduler.run(({ cold, expectObservable }) => {
+      const test$ = cold('a-b-c-|', { a: 1, b: 2, c: 3 });
+
+      expectObservable(
+        subject.pipe(
+          // convert error to "error" to match expectation of `#` in marble diagram
+          catchError((err) =>
+            err instanceof BufferedReducerStopException ? throwError(() => 'error') : throwError(() => err)
```

Comment:
`throwError` is overkill for this case (usually for when a subject or high order observable unwrapping, etc), you can just regular `throw`. And since this is just testing, it doesn't even need to be an `Error` type. 

bonus fun fact: `throwError` takes a factory OR the thing to throw, so `throwError(new Error('beep'))` or even `throwError('error')` would have sufficed

```suggestion
            err instanceof BufferedReducerStopException ? throw 'error') : throw err
```

Reply:
Apparently passing a value to `throwError` is deprecated, the factory is the recommended way.
But yeah, overkill is fair.

Reply:
https://github.com/ReactiveX/rxjs/blob/7.8.1/src/internal/observable/throwError.ts#L104

well i'll be, makes sense. thanks!